src/DeviceInterface/InterfacePrototype.js

```diff
@@ -806,7 +806,17 @@ class InterfacePrototype {
                 password: this.passwordGenerator.password,
                 username: this.emailProtection.lastGenerated,
             });
-            this.storeFormData(formData, 'formSubmission');
+
+            // If credentials has only username field, and no password field, then trigger is a partialSave
+            const isUsernameOnly =
+                Boolean(formData.credentials?.username) && !formData.credentials?.password && [...form.inputs.credentials].length === 1;
```

Comment:
Isn't this a `Set`? In that case, you don't need to destructure, you can just call [`.size`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/size). Same below, and if you know of any other instance where we're destructuring just to get the size, please update it.